app/lib/classification-queue.js

```diff
@@ -0,0 +1,101 @@
+import apiClient from 'panoptes-client/lib/api-client';
+import { Split } from 'seven-ten';
+
+const FAILED_CLASSIFICATION_QUEUE_NAME = 'failed-classifications';
+const MAX_RECENTS = 10;
+
+class ClassificationQueue {
+    constructor(storage, api) {
+        this.storage = storage || window.localStorage;
+        this.apiClient = api || apiClient;
+        this.recents = [];
+    }
+
+    add(classification) {
+        var queue = this._loadQueue();
+        queue.push(classification);
+
+        try {
+            this._saveQueue(queue);
+            console.info('Queued classifications:', queue.length);
+        }
+        catch(error) {
+            console.error('Failed to queue classification:', error);
+        }
+
+        this.flushToBackend();
+    }
+
+    length() {
+        return this._loadQueue().length
+    }
+
+    addRecent(classification) {
+        if (this.recents.length > MAX_RECENTS) {
+            let droppedClassification = this.recents.pop();
+            droppedClassification.destroy();
+        }
+
+        this.recents.unshift(classification);
+    }
+
+    flushToBackend() {
+        var queue = this._loadQueue();
+
+        if (queue.length > 0) {
+            console.log('Saving queued classifications:', queue.length);
+            for (let classificationData of queue) {
```

Comment:
`for…of` isn’t supported by arrays in IE. I think we’ve preferred `Array.forEach` for better browser compatibility so `queue.forEach( function(classificationData) {…})` might be safer here.

Reply:
Seems to work in IE11 on BrowserStack, so I might be too paranoid about browser compatibility here.

Reply:
`const` instead of `let` here.

Reply:
There's a `babel-plugin-transform-es2015-for-of` in the package.lock. Might be that?

Reply:
Could well be.